sherpa/astro/io/tests/test_io.py

```diff
@@ -416,6 +416,36 @@ def test_fits_file_missing_column(make_data_path):
         io.read_table(infile, colkeys=["ra", "Foo"])
 
 
+@requires_fits
+@requires_data
+def test_get_header_data_missing_key(make_data_path):
+    """What happens if a requested key is missing?
+
+    TODO: If get_header_data is useful should we export if from
+    the io, and not backend, level?
+    """
```

Comment:
Good idea~ I think it would be useful.

Reply:
I've made #2071 so we can discuss that there.